accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,340 @@
+
+The IAL is an abstraction that is internal to the .NET SDK.  It is not intended as a public extensibility mechanism to allow supporting arbitrary installers, and we may make breaking changes to its API as needed.
+
+# Supported installation types
+
+We plan to (eventually) support the following installation types.  Some of them may not be supported in .NET 6 if we don't have workloads in .NET 6 that are supported on the corresponding platforms.
+
+## .NET SDK managed installation
+
+In this implementation of the IAL, the .NET SDK itself manages workload installations directly by writing files to the dotnet folder.  This is generally for copies of the .NET SDK which were not installed via an OS installer, for example those that are installed via the dotnet-install scripts, or via unzipping archives of the .NET SDK.
+
+Other terms we have at times used for this installation type: xcopy, universal installer, files on disk, NuGet installer
+
+## Visual Studio managed installation
+
+This is an installation of the .NET SDK installed and managed by the Visual Studio installer.  The .NET SDK workloads will have corresponding Visual Studio workloads or components, which will install workload packs via Windows MSIs.  Note that the Visual Studio workloads or components should also include the necessary Visual Studio components (designers, project system, etc) to support the workload, in addition to the .NET SDK workload packs.
+
+## Standalone Windows installation
+
+This is an installation of the .NET SDK installed on Windows via the .NET SDK installer bundle, which is the .exe which can be downloaded on the .NET SDK download pages.  The bundle installs the same MSIs for the .NET SDK as Visual Studio does, and the workload installer will likewise install the same MSIs for workload packs as the Visual Studio installer.
+
+The same dotnet folder may have different versions of the SDK installed by both the Visual Studio installer and the Standalone installer, which may share the same versions of some components.  For the .NET SDK, the underlying MSIs are ref-counted so that they are uninstalled when there are no more installations that depend on them.  We will need to make sure that workload installation and management also works well when there are different install types coexisting.
+
+## Package manager installation
+
+If the .NET SDK is installed via a package manager such as APT or RPM, we may want to use the same package manager to install workloads.  However, we are not currently sure if this is necessary.  Our initial impression is that it may be OK to use a package manager to install the core .NET SDK, but then have the SDK itself manage the workloads (ie via the .NET SDK managed IAL implementation).
+
+For .NET 6, we will simply use the .NET SDK managed IAL, and gather feedback on whether we need workloads to be more integrated with package managers in a future release.
+
```

Comment:
We have some experience with this in source-build .NET.

For .NET Core 1.x and 2.x, ASP.NET Core was not in a good source-buildable state. So for those versions, a source-built SDK would consume ASP.NET from nuget.org packages.

Even though our packages didn't contain ASP.NET Core, the SDK still tracked what was the latest version of ASP.NET Core. This meant that using the latest SDK package would ensure also the ASP.NET Core security patches were used. When we push new versions of our .NET Core SDK container images, CD flows use this as a trigger rebuild apps, which ensures the security fixes are deployed.

The main downside for users was the additional time spent on each build to download the ASP.NET packages. There is no caching of these packages because each container build starts from a fresh SDK image.